src/MetadataStorage.ts

```diff
@@ -86,25 +87,36 @@ export class MetadataStorage {
   public firestoreRef: Firestore = null;
 }
 
+/**
+ * Return exisiting metadataStorage, otherwise create if not present
+ */
 export const getMetadataStorage = (): MetadataStorage => {
-  if (!store) {
-    throw new Error(
-      'Application has not been initialized. Call Initialize() method'
-    );
+  const store = getGlobalStore();
+
+  if (!store.metadataStorage) {
+    initializeMetadataStorage();
   }
 
   return store.metadataStorage;
 };
 
-export const Initialize = (
-  firestore: Firestore,
-  metadataStore: IMetadataStore = globalStore
-): void => {
-  store = metadataStore;
+function initializeMetadataStorage() {
+  const store = getGlobalStore();
 
   if (!store.metadataStorage) {
     store.metadataStorage = new MetadataStorage();
   }
+}
+
+/**
+ * Used for testing to reset metadataStore to clean state
+ */
+export function clearMetadataStore() {
```

Comment:
This function actually clears the metadataStorage so `clearMetadataStorage` is a better name